### PR TITLE
fix warning mangles output format

### LINKED ISSUE #8062: warnings in output destroys json encoding
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | e.g. 0.17.2            |

when the analysis issues warnings like

grafik

these scramble the json output when using `--format=json`

I think we should either send the output on stderr instead of stdout, or need a proper formatted error according to the used `--format`

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/rectorphp/rector/issues/8062

### CODE REVIEW

**Comment by staabm** [comment]
the failling test works, the fix fixes the test, but the warning is lost somewhere in rector … :-)

seems stderr is not yet used at all in the rector console app?

File: src/Application/FileProcessor/PhpFileProcessor.php [hunk 1]
```diff
@@ -96,7 +96,7 @@ public function process(File $file, Configuration $configuration): array
 
         // show warning on has InlineHTML node if file has changed
         if ($fileHasChanged && $file->hasInlineHTMLNode()) {
-            $this->symfonyStyle->warning(sprintf(
+            $this->symfonyStyle->getErrorStyle()->warning(sprintf(
```
Comment (samsonasik) [inline src/Application/FileProcessor/PhpFileProcessor.php:99]: For console format, expected output should show warning, see https://github.com/rectorphp/rector-src/actions/runs/5569673660/jobs/10173190723#step:11:17

you may probably need to have instanceof check for formatter.

Comment (staabm) [reply src/Application/FileProcessor/PhpFileProcessor.php:99]: Sure. But this means the warning only shows up on default cli usage and not when using any other output format. 

IMO stderr in rector is broken, but the instanceof is maybe good enough for now.

**Comment by staabm** [comment]
I think the current architecture of worker processes, output format handling etc. doesn't allow to fix the issue in this way.

File: src/Application/FileProcessor/PhpFileProcessor.php (hunk 1 above)
Comment (staabm) [reply src/Application/FileProcessor/PhpFileProcessor.php:99]: would it make sense to turn this warning into a system-error?

it looks like its the only warning which is outputted by a worker process this way atm and IMO it violates existing rector process managment

**Comment by TomasVotruba** [comment]
I think it would be more straighforward to handle this in returned array values, as all other errors. E.g. add "warnings" section.

**Pull request closed without merging**